### PR TITLE
[HLRC] Add Start/Stop Watch Service APIs.

### PR DESCRIPTION
**Pull request by jdconrad**
This adds both the start water service and stop watcher service apis to the HLRC.  Includes new request objects, tests, and documentation.

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StartWatchServiceRequest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * under the License.
+ */
+
+package org.elasticsearch.client.watcher;
+
+import org.elasticsearch.client.Validatable;
+
+public class StartWatchServiceRequest implements Validatable {
+
+    public StartWatchServiceRequest() {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StartWatchServiceRequest.java]: Do we need default constructors here?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StopWatchServiceRequest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * under the License.
+ */
+
+package org.elasticsearch.client.watcher;
+
+import org.elasticsearch.client.Validatable;
+
+public class StopWatchServiceRequest implements Validatable {
+
+    public StopWatchServiceRequest() {
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StopWatchServiceRequest.java]: default constructor?

File: docs/java-rest/high-level/watcher/start-watch-service.asciidoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
+[[java-rest-high-watcher-start-watch-service]]
```
Comment (hub-cap) [inline docs/java-rest/high-level/watcher/start-watch-service.asciidoc]: These have a new tidier format. See #34203 for a *much* more concise version of the docs. Pls do this for both of the asciidoc pages you have added

**Review by hub-cap — APPROVED** [review]
Easy review to do. Thanks muchos. Please just update the docs and consider (nit) taking out those constructors. Im going to approve this, assuming you will be DRY'ing the docs pages and running the docs build locally as well. Ping me if you want a re-review.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StartWatchServiceRequest.java (hunk 1 above)
Comment (jdconrad) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StartWatchServiceRequest.java]: Removed.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StopWatchServiceRequest.java (hunk 2 above)
Comment (jdconrad) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/StopWatchServiceRequest.java]: Removed.

File: docs/java-rest/high-level/watcher/start-watch-service.asciidoc (hunk 3 above)
Comment (jdconrad) [reply docs/java-rest/high-level/watcher/start-watch-service.asciidoc]: Done.

**Comment by jdconrad** [comment]
@hub-cap Thanks for the reivew!  Will commit once CI passes.

**Pull request merged**